cli/src/pcluster/templates/pcluster_dashboard_builder.py

```diff
@@ -0,0 +1,586 @@
+        self.graph_height = 6
+        self.logs_width = 24
+        self.logs_height = 6
+        self.empty_section = {"bool": True}
+
+        self._add_resources()
+
+    # -- Utility methods --------------------------------------------------------------------------------------------- #
+
+    @property
+    def _stack_region(self):
+        return core.Stack.of(self).region
+
+    # -- Resources --------------------------------------------------------------------------------------------------- #
+
+    def _add_resources(self):
+
+        # Initialize a cloudwatch dashboard
+        cloudwatch_dashboard = cloudwatch.Dashboard(
+            scope=self.stack_scope, id="CloudwatchDashboard", dashboard_name=self.dashboard_name
+        )
+
+        # Create a text widget for title "Head Node EC2 metrics"
+        self._add_text_widget("\n# Head Node EC2 Metrics\n", cloudwatch_dashboard)
+
+        # Head node instance graphs
+        self._add_head_node_instance_metrics_graphs(cloudwatch_dashboard)
+
+        ebs_metrics = [
+            {"metrics": ["VolumeReadOps", "VolumeWriteOps"], "title": "Read/Write Ops"},
+            {"metrics": ["VolumeReadBytes", "VolumeWriteBytes"], "title": "Read/Write Bytes"},
+            {"metrics": ["VolumeTotalReadTime", "VolumeTotalWriteTime"], "title": "Total Read/Write Time"},
+            {"metrics": ["VolumeQueueLength"], "title": "Queue Length"},
+            {"metrics": ["VolumeIdleTime"], "title": "Idle Time"},
+        ]
+        conditional_ebs_metrics = [
+            {
+                "metric": "VolumeConsumedReadWriteOps",
+                "supported_vol_types": ["io1", "io2", "gp3"],
+                "title": "Consumed Read/Write Ops",
+            },
+            {
+                "metric": "VolumeThroughputPercentage",
+                "supported_vol_types": ["io1", "io2", "gp3"],
+                "title": "Throughput Percentage",
+            },
+            {"metric": "BurstBalance", "supported_vol_types": ["gp2", "st1", "sc1"], "title": "Burst Balance"},
+        ]
+
+        # EBS metric graphs
+        number_of_ebs_volumes = len(self._storage_resource_ids[SharedStorageType.EBS])
+        if number_of_ebs_volumes > 0:
+            self._add_ebs_metrics_graphs(
+                cloudwatch_dashboard=cloudwatch_dashboard,
+                ebs_metrics=ebs_metrics,
+                conditional_ebs_metrics=conditional_ebs_metrics,
+            )
+
+        # Raid metric graphs
+        number_of_raid_volumes = len(self._storage_resource_ids[SharedStorageType.RAID])
+        if number_of_raid_volumes > 0:
+            self._add_raid_metrics_graphs(
+                cloudwatch_dashboard=cloudwatch_dashboard,
+                ebs_metrics=ebs_metrics,
+                conditional_ebs_metrics=conditional_ebs_metrics,
+            )
+
+        # EFS metrics graphs
+
+        number_of_efs_volumes = len(self._storage_resource_ids[SharedStorageType.EFS])
+        if number_of_efs_volumes > 0:
```

Comment:
minor: you can avoid defining the number_of_xxx_volumes variable, because you're usign it in a single place.
YOu can do:
```
        if len(self._storage_resource_ids[SharedStorageType.EFS]) > 0:
```

Reply:
I was putting them into separate lines now because I think the logic of `number_of_efs_volumes` will be changed to len(storage_list) after I get the list of storage objects. So it will be easy to changed in the future